functions/fish_prompt.fish

```diff
@@ -41,7 +43,11 @@ function fish_prompt
     echo -n -s " " $directory_color $cwd $normal_color
     echo -n -s " on " $repository_color (git_branch_name) $normal_color " "
 
-    if git_is_touched
+    if git_is_touched; and test "$theme_stash_indicator" = 'yes'; and git_is_stashed
+      echo -n -s $stash " " $dirty
+    else if test "$theme_stash_indicator" = 'yes'; and git_is_stashed
+      echo -n -s $stash
+    else if git_is_touched
```

Comment:
I'd say this is readable and explicit. I don't think simplifying would make it more readable/maintainable. Some repetition is good repetition.

You have any ideas, @faho?

Reply:
The issue is `git_is_touched` calls `git diff-index` and `git diff`, which are fairly expensive as they need to scan the entire repo. This potentially calls them twice.

I'd probably do it like this:

```fish
if test "$theme_stash_indicator" = yes; and git_is_stashed
  if git_is_touched
    echo -n -s $stash " " $dirty
  else
    echo -n -s $stash
  end
else if git_is_touched
  # ...
end
```

Alternatively sorting the first if-condition as `test "$theme_stash_indicator" = yes; and git_is_stashed; and git_is_touched` would at least not call that twice, but could still check `git_is_stashed` twice. It's cheaper, but better not to do that work if you don't need to.

Reply:
Alternatively alternatively, you'd build up the string to `echo` and then print it in one go:

```fish
set -l list
if test "$theme_stash_indicator" = yes; and git_is_stashed
    set list $list $stash
end
if git_is_touched
    set list $list $dirty
end

echo -n $list
```

Reply:
perfect, the latter is what i was trying to do initially but i didn't think to look at what `-s` does 🤦🏽‍♂️ 

Updated the PR and here's one more screenshot to verify, it works the same as the last version:

<img width="843" alt="Screen Shot 2022-02-25 at 12 14 43" src="https://user-images.githubusercontent.com/3088615/155793628-dea86f82-f045-456b-8405-c9813ef5fb15.png">